tsl/src/continuous_aggs/create.c

```diff
@@ -316,6 +324,42 @@ cagg_add_trigger_hypertable(Oid relid, char *trigarg)
 	ts_cache_release(hcache);
 }
 
+/* add additional indexes to materialization table for the columns derived from
+ * the group-by column list of the partial select query
+ * if partial select query has:
+ * GROUP BY timebucket_expr, <grpcol1, grpcol2, grpcol3 ...>
+ * index on mattable is <grpcol1, timebucketcol>, <grpcol2, timebucketcol> ... and so on.
+ * i.e. #indexes =(  #grp-cols - 1)
+ */
+static void
+mattablecolumninfo_add_mattable_index(MatTableColumnInfo *matcolinfo, Hypertable *ht)
+{
+	IndexStmt stmt = {
+		.type = T_IndexStmt,
+		.accessMethod = DEFAULT_INDEX_TYPE,
+		.idxname = NULL,
+		.relation = makeRangeVar(NameStr(ht->fd.schema_name), NameStr(ht->fd.table_name), 0),
+		.tableSpace = get_tablespace_name(get_rel_tablespace(ht->main_table_relid)),
+	};
+	IndexElem timeelem = { .type = T_IndexElem,
+						   .name = matcolinfo->matpartcolname,
+						   .ordering = SORTBY_DESC };
+	ListCell *le = NULL;
+	foreach (le, matcolinfo->mat_groupcolname_list)
+	{
+		char *grpcolname = (char *) lfirst(le);
```

Comment:
foreach index created we should elog a NOTICE `adding group index named %s BTREE(%s, %s)`